codec/reflectcodec/type_codec.go

```diff
@@ -19,6 +19,12 @@ import (
 // DefaultTagName that enables serialization.
 const DefaultTagName = "serialize"
 
+var (
+	MagicBytesPtrNil        = []byte{0xff, 0x32, 0x31, 0x36, 0x98}
+	MagicBytesPtr           = []byte{0xc3, 0xa1, 0x00, 0x33, 0x25}
+	magicBytesPtrPrefixSize = len(MagicBytesPtrNil)
+)
```

Comment:
I don't think we should be using this. My hope with using `omitempty` was to _always_ include a boolean indicating if the pointer / interface was included or not. Fuzzy matching here (in a consensus critical codec) seems like a nightmare

Reply:
I agree, this bit of code was added _before_ the `omitempty` (which is opt-in). I will make it a single byte.